core/sqlx_parser.ts

```diff
@@ -0,0 +1,257 @@
+  SINGLE_LINE_COMMENT: LEXER_STATE_NAMES.SQL + "_singleLineComment",
+  MULTI_LINE_COMMENT: LEXER_STATE_NAMES.SQL + "_multiLineComment",
+  SINGLE_QUOTE_STRING: LEXER_STATE_NAMES.SQL + "_singleQuoteString",
+  DOUBLE_QUOTE_STRING: LEXER_STATE_NAMES.SQL + "_doubleQuoteString",
+  START_JS_PLACEHOLDER: LEXER_STATE_NAMES.SQL + "_startJsPlaceholder",
+  ESCAPED_BACKTICK: LEXER_STATE_NAMES.SQL + "_escapedBacktick",
+  BACKTICK: LEXER_STATE_NAMES.SQL + "_backtick",
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.SQL + "_captureEverythingElse"
+};
+
+const JS_BLOCK_LEXER_TOKEN_NAMES = {
+  SINGLE_LINE_COMMENT: LEXER_STATE_NAMES.JS_BLOCK + "_singleLineComment",
+  MULTI_LINE_COMMENT: LEXER_STATE_NAMES.JS_BLOCK + "_multiLineComment",
+  SINGLE_QUOTE_STRING: LEXER_STATE_NAMES.JS_BLOCK + "_singleQuoteString",
+  DOUBLE_QUOTE_STRING: LEXER_STATE_NAMES.JS_BLOCK + "_doubleQuoteString",
+  START_JS_TEMPLATE_STRING: LEXER_STATE_NAMES.JS_BLOCK + "_startJsTemplateString",
+  START_JS_BLOCK: LEXER_STATE_NAMES.JS_BLOCK + "_startJsBlock",
+  CLOSE_BLOCK: LEXER_STATE_NAMES.JS_BLOCK + "_closeBlock",
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.JS_BLOCK + "_captureEverythingElse"
+};
+
+const JS_TEMPLATE_STRING_LEXER_TOKEN_NAMES = {
+  ESCAPED_BACKSLASH: LEXER_STATE_NAMES.JS_TEMPLATE_STRING + "_escapedBackslash",
+  ESCAPED_BACKTICK: LEXER_STATE_NAMES.JS_TEMPLATE_STRING + "_escapedBacktick",
+  ESCAPED_DOLLAR_BRACE: LEXER_STATE_NAMES.JS_TEMPLATE_STRING + "_escapedDollarBrace",
+  START_JS_BLOCK: LEXER_STATE_NAMES.JS_TEMPLATE_STRING + "_startJsBlock",
+  CLOSE_STRING: LEXER_STATE_NAMES.JS_TEMPLATE_STRING + "_closeString",
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.JS_TEMPLATE_STRING + "_captureEverythingElse"
+};
+
+const INNER_SQL_BLOCK_LEXER_TOKEN_NAMES = {
+  STATEMENT_SEPERATOR: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_statementSeparator",
+  SINGLE_LINE_COMMENT: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_singleLineComment",
+  MULTI_LINE_COMMENT: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_multiLineComment",
+  SINGLE_QUOTE_STRING: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_singleQuoteString",
+  DOUBLE_QUOTE_STRING: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_doubleQuoteString",
+  START_JS_BLOCK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_startJsBlock",
+  CLOSE_BLOCK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_closeBlock",
+  ESCAPED_BACKTICK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_escapedBacktick",
+  BACKTICK: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_backtick",
+  CAPTURE_EVERYTHING_ELSE: LEXER_STATE_NAMES.INNER_SQL_BLOCK + "_captureEverythingElse"
+};
+
+const lexer = moo.states(buildSqlxLexer());
+
+export interface ISqlxParseResults {
+  config: string;
+  js: string;
+  sql: string[];
+  incremental: string;
+  preOperations: string[];
+  postOperations: string[];
+}
+
+export function parseSqlx(code: string): ISqlxParseResults {
+  const results = {
+    config: "",
+    js: "",
+    sql: [""],
+    incremental: "",
+    preOperations: [""],
```

Comment:
What's the reason for this initialization?

Reply:
discussed on chat, hopefully makes sense now